apps/gp2-server/src/controllers/user.controller.ts

```diff
@@ -20,26 +27,31 @@ export default class Users implements UserController {
   async update(
     id: string,
     update: gp2.UserUpdateRequest,
+    currentUserId: string,
   ): Promise<gp2.UserResponse> {
     await this.userDataProvider.update(id, update);
-    return this.fetchById(id);
+    return this.fetchById(id, currentUserId);
   }
 
   async fetch(options: gp2.FetchUsersOptions): Promise<gp2.ListUserResponse> {
     const { total, items: users } = await this.userDataProvider.fetch(options);
 
-    const items = total > 0 ? users.map(parseUserToResponse) : [];
+    const items =
+      total > 0 ? users.map((user) => parseUserToResponse(user)) : [];
```

Comment:
that's the same as before no?

Reply:
No. parseUserToResponse now takes 2 parameters the user, and the current user id.

Like this users.map(parseUserToResponse), parseUserToResponse will be called with the user as the first param, and the index as the second.

i.e. the parameters of the callback. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map